server/channels/testlib/helper.go

```diff
@@ -41,13 +41,23 @@ type HelperOptions struct {
 }
 
 func NewMainHelper() *MainHelper {
+	// Ignore any globally defined datasource if a test dsn defined
+	if os.Getenv("TEST_DATABASE_MYSQL_DSN") != "" || os.Getenv("TEST_DATABASE_POSTGRESQL_DSN") != "" {
+		os.Unsetenv("MM_SQLSETTINGS_DATASOURCE")
+	}
```

Comment:
Given the simplifications now, is it feasible to converge on CI just setting `MM_SQLSETTINGS_DATASOURCE` and the main helper using it to create the new database and then unilaterally unsetting it so that the tests use the configured one only? (i.e. one less moving part)

Reply:
I think it's feasible and would be a great improvement. I found some issues when I tried to do it at first, and decided to leave it out of this PR not to make it bigger, but I think we should work on it as a follow up